repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/powerfilter/GraphiteTest.groovy

```diff
@@ -67,12 +70,13 @@ class GraphiteTest extends ReposeValveTest {
         def mc1 = deproxy.makeRequest(url: reposeEndpoint + "/endpoint")
         def mc2 = deproxy.makeRequest(url: reposeEndpoint + "/endpoint")
         def mc3 = deproxy.makeRequest(url: reposeEndpoint + "/cluster")
-        sleep(2000)
 
         then:
         mc1.receivedResponse.code == "200"
         mc2.receivedResponse.code == "200"
         mc3.receivedResponse.code == "200"
-        lastCount == 3
+        waitForCondition(new SystemClock(), '2s', '250', {
```

Comment:
Being that this is a `ReposeValveTest`, you have access to a `SystemClock` instance via the `repose.clock` (i.e., from the `ReposeValveLauncher` instance). It doesn't make much difference, but it may be preferable to use that clock instance rather than creating a new one.

In fact, we should probably update `ReposeValveLauncher` to use the static `SystemClock.instance()` or `SystemClock.INSTANCE` rather than creating a new instance ourselves.